### PR TITLE
Deprecate Script Scan Rules and related classes

### PR DESCRIPTION
**Pull request by ricekot**
These will be made available as part of the scripts add-on.

### CODE REVIEW

File: zap/src/main/java/org/zaproxy/zap/control/CoreFunctionality.java [hunk 1]
```diff
@@ -142,24 +141,7 @@ private static synchronized void createExtensions() {
      * @see AbstractPlugin
      */
     public static List<AbstractPlugin> getBuiltInActiveScanRules() {
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/control/CoreFunctionality.java:143]: Remove it?

Comment (ricekot) [reply zap/src/main/java/org/zaproxy/zap/control/CoreFunctionality.java:143]: Kept it because it was public, but I don't mind removing it.

Comment (thc202) [reply zap/src/main/java/org/zaproxy/zap/control/CoreFunctionality.java:143]: I think we can consider this class internal, it's to bootstrap ZAP. We can also deprecate it first.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by ricekot** [comment]
Set to WIP because this should only be merged after https://github.com/zaproxy/zap-extensions/pull/5175.

**Review by kingthorin — APPROVED** [review]

**Comment by thc202** [comment]
@ricekot ok to merge now?

**Pull request merged**